### PR TITLE
Fix share confirm modal appearing on ideas button

### LINKED ISSUE #6683: Ideas page button triggers share conformation
### Description

When the "Choose a tutorial" button is pressed on [the ideas page](https://scratch.mit.edu/ideas), the "Are you sure you want to share?" popup appears.

### Steps to reproduce

1. Turn on the Conformations addon
2. Open https://scratch.mit.edu/ideas
3. Click on the "Choose a tutorial" button

### Demonstration(s) and additional context

_No response_

### Environment

Browser: Firefox 117
Scratch Addons version: 1.34.1
Operating system: macOS Ventura

### PR DESCRIPTION
**Pull request by mybearworld**
Resolves #6683

### Changes

Make the `.banner-button` selector more specific by also requiring a `.banner-text` next to it, which the ideas button doesn't have.

### Reason for changes

To fix the share modal appearing on the ideas button.

### Tests

Tested on Edge 116.

### CODE REVIEW

**Comment by WorldLanguages** [comment]
> Make the `.banner-button` selector more specific by also requiring a `.banner-text` next to it, which the ideas button doesn't have.

Maybe we can change it to something even more specific to the Share button? Is there anything?

**Comment by mxmou** [comment]
> > Make the `.banner-button` selector more specific by also requiring a `.banner-text` next to it, which the ideas button doesn't have.
> 
> Maybe we can change it to something even more specific to the Share button? Is there anything?

`.preview .banner-button` would match a `.banner-button` on the project page, but there's no "share banner" or "share button" class.

**Comment by mybearworld** [comment]
> > Make the `.banner-button` selector more specific by also requiring a `.banner-text` next to it, which the ideas button doesn't have.
> 
> Maybe we can change it to something even more specific to the Share button? Is there anything?



> > Make the `.banner-button` selector more specific by also requiring a `.banner-text` next to it, which the ideas button doesn't have.
> 
> Maybe we can change it to something even more specific to the Share button? Is there anything?

It's very generic.

```html
<div class="banner-outer">
  <div class="flex-row inner banner-inner">
    <span class="banner-text">
      <span>
        This project is not shared — so only you can see it. Click share to let
        everyone see it!
      </span>
    </span>
    <button class="button banner-button"><span>Share</span></button>
  </div>
</div>
```

Something with `:has` would probably work, but that doesn't have good enough browser support.

**Comment by MaterArc** [comment]
It seems scratch got a little lazy in naming their elements...

**Comment by DNin01** [comment]
> It seems scratch got a little lazy in naming their elements...

They haven't really. The Scratch source code has more access to itself than a browser extension and knows variables the extension doesn't. A separate, unrelated module like our addon's script has to work with limited information (especially since Scratch uses things like [React](https://github.com/facebook/react)) and therefore may have trouble distinguishing a certain button from others. This is why updates to the editor can leave added buttons in the wrong place.

The parent elements provide more contextual information - selecting `.preview .banner-button` as pointed out by @mxmou should solve the problem. That or checking the URL.

**Comment by WorldLanguages** [comment]
Adding `body:not(.sa-body-editor)` to the beginning of the selector would be pretty much equivalent to checking that the URL is a project page.

**Comment by mybearworld** [comment]
> Adding `body:not(.sa-body-editor)` to the beginning of the selector would be pretty much equivalent to checking that the URL is a project page.

This seems to only get added when you're in the editor.

**Comment by WorldLanguages** [comment]
> Adding `body:not(.sa-body-editor)` to the beginning of the selector would be pretty much equivalent to checking that the URL is a project page.

Response to myself: I don't think it is very useful in this case, though.

**Review by WorldLanguages — APPROVED** [review]
Works!

**Review by Samq64 — APPROVED** [review]

File: addons/confirm-actions/userscript.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ export default async function ({ addon, console, msg }) {
       let cancelMessage = null;
       if (
         addon.settings.get("projectsharing") &&
-        e.target.closest("[class*='share-button_share-button']:not([class*='is-shared']), .banner-button")
+        e.target.closest(
+          "[class*='share-button_share-button']:not([class*='is-shared']), .banner-text + .banner-button"
```
Comment (DNin01) [inline addons/confirm-actions/userscript.js:17]: I would have also selected `.preview` but this is good too.

**Review by DNin01 — APPROVED** [review]

**Pull request merged**